Project1/Grid.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+#include <functional>
+#include <map>
+#include <vector>
+#include <list>
+#include <queue>
+#include "CellPath.h"
+#include "Party.h"
+#include "ID.h"
+/**
+Grid class. Holds a Matrix of GridCells. Should be initialized at the start of each chapter.
+*/
+class GridCell;
+class Unit;
+class Mobility;
+
+class Grid {
+public:
+	//vector wrapper that only allows [] op
+	template<typename VecCell>
+	class grid_row {
+	private:
+		VecCell& _row;
+		grid_row(VecCell& row) : _row(row) {}
+	public:
+		auto& operator[](size_t index) { return _row[index];}
+		const auto& operator[](size_t index) const { return _row[index];}
+		friend class Grid;
+	};
+	using GridRow = grid_row < std::vector<GridCell>>;
+	using constGridRow = grid_row<const std::vector<GridCell>>;
+private:
+
+	std::map<const Unit*, GridCell*> _unit_to_cell;
+	std::map<const GridCell*, Unit*> _cell_to_unit;
+
+	void insertAdjacencies();
+	
+
+protected:
+	std::vector<std::vector<GridCell>> _grid;
+
+	Grid(int width, int height);
+	Grid(int width, int height, std::vector<PartyData> data);
+	~Grid() = default;
+
+	Unit* getUnit(const GridCell& index);
+	const Unit* getUnit(const GridCell& index) const;
+public:
+	
+	std::list<Party> _parties;
+
+	Grid(const Grid& map) = delete;
+	Grid& operator=(const Grid & map) = delete;
+	Grid(Grid&& map) = default;
+	Grid& operator=(Grid&& map) = default;
+		
+	void insertParty(PartyData data);
+	void moveUnit(GridCell& start, GridCell& destination);
+	void insertUnit(Unit& new_unit, GridCell& destination);
+	void removeUnit(Unit& unit);
+	
+	bool hasUnit(const GridCell & index) const;
+	bool hasUnit(const Unit & unit) const;
+
+	Unit& operator[](const GridCell& index);
```

Comment:
I think this should be a named method that takes an ID

Probably all of the public methods that take `const GridCell&` you should consider eitther taking an ID instead or overloading for ID.